apps/web/src/components/ShareDialog.tsx

```diff
@@ -0,0 +1,97 @@
+// ShareDialog.tsx
+import React, { useState, useEffect } from "react";
+import * as Dialog from "@radix-ui/react-dialog";
+import * as Switch from "@radix-ui/react-switch";
+import { FiShare } from "react-icons/fi";
+import { BiLink } from "react-icons/bi";
+import { AiOutlineCheck } from "react-icons/ai";
+import { TwitterIcon, TwitterShareButton } from "react-share";
+import { ShareToLens, Theme, Size } from "@lens-protocol/widgets-react";
+export type ShareDialogProps = {
+  title: string;
+  content: string;
+};
+
+export const ShareDialog: React.FC<ShareDialogProps> = ({ title, content }) => {
+  const [uniqueURL, setUniqueURL] = useState("");
+  const [pageTitle, setPageTitle] = useState("");
+  const [pageContent, setPageContent] = useState("");
+  const [isCopied, setIsCopied] = useState(false);
+
+  const copyToClipboard = () => {
+    navigator.clipboard.writeText(uniqueURL);
+    setIsCopied(true);
+  };
+  useEffect(() => {
+    setPageTitle(title);
+    setPageContent(content);
+    setUniqueURL(window.location.href);
+  }, [content, title, uniqueURL]);
+
+  return (
+    <Dialog.Root>
```

Comment:
✨  Could we use [`Popover`](https://www.radix-ui.com/docs/primitives/components/popover) here instead? Then we don't have to handle the positioning manually!

Reply:
If we want is as this design. https://www.figma.com/file/ErWyhBi4kGxGqNzQxX7uPL/denoted?t=JZH1wbI1lXrXuYuh-0

Then we still need to move the `Popover.Content` manually. Otherwise it will be situated right under like so: 
<img width="294" alt="image" src="https://user-images.githubusercontent.com/59825171/232235836-49812d10-8d33-49d0-bf94-f5d4ae0f58fd.png">

Maybe I'm missing something?

Reply:
We can use the `align` property for that! See here:
text-davinci-003